### DISCUSSION TITLE
CWL compliance level?

### DISCUSSION THREAD

**Discussion by buddyroo30** [lanl/BEE#681] [Q&A] [answered]
It isn't clear exactly what BEE's CWL compliance is (except it says it supports a subset of CWL). Is there any more info about this? E.g. is it CWL 1.0, CWL 1.1, or CWL 1.2 compliant? Thanks.

**Comment by Boogie3D** [comment, marked as answer]
BEE is a partial implementation of CWL v1.0 that supports the Workflow and CommandLineTool classes, all data types except `record`, basic input and output binding functionality, stdout and stderr capture, glob, etc. We currently do not support JavaScript or any requirements besides `DockerRequirement`. As this implementation is geared toward HPC use cases, our `DockerRequirement` implementation is non-standard and does not actually support Docker directly, but rather [Charliecloud](https://hpc.github.io/charliecloud/). As we do not support JavaScript, our implementation of expressions is also non-standard and is limited to input value reference and concatenation. Our [CLAMR-FFMPEG](https://github.com/lanl/BEE/tree/main/examples/clamr-ffmpeg-build) example gives a good idea of what we currently support.
